test/Ansatz/Chain_test.jl

```diff
@@ -107,6 +107,36 @@
     @testset "Canonization" begin
         using Tenet
 
+        @testset "contract" begin
+            qtn = rand(Chain, Open, State; n = 5, p = 2, χ = 20)
+            canonized = canonize(qtn)
+
+            @test_throws ArgumentError contract(canonized, :between, Site(1), Site(2); direction = :dummy)
+            @test_throws ArgumentError contract!(canonized, :between, Site(1), Site(2); direction = :dummy)
+
+            canonized = canonize(qtn)
```

Comment:
You got it repeated
```suggestion
```

Reply:
This is necessary because we have the in place contraction

Reply:
Ahh okay, makes sense. But then use the `let ... end` please.